handlers/api.go

```diff
@@ -1121,37 +1121,60 @@ func getSyncCommitteeSlotsStatistics(validators []uint64, epoch uint64) (types.S
-		// a sync committee takes abouth 27h so in the span of a day we migh have a maximum of two different sync committees (one being unfinished)
-		// to lower the bigtable workload, we only check for validators that are querried AND in the current or the last sync committee
-		periods := []int64{int64(utils.SyncPeriodOfEpoch(epoch))}
-		if periods[0] > 0 {
+		// get relevant period
+		periodOfEpoch := utils.SyncPeriodOfEpoch(epoch)
+		periods := []int64{int64(periodOfEpoch)}
+		firstEpochOfPeriod := utils.FirstEpochOfSyncPeriod(periodOfEpoch)
+		// if the committee period before the relevant one is also not yet fully exported, add it to the query
+		if periods[0] > 0 && lastExportedEpoch < firstEpochOfPeriod-1 {
 			periods = append(periods, periods[0]-1)
 		}
 
-		var validatorsInSyncCommittees struct {
+		// get all validators part of the relevant committees, grouped by period
+		var syncCommitteeValidators []struct {
 			Validators pq.Int64Array `db:"validators"`
 		}
-		query, args, err := sqlx.In(`SELECT COALESCE(ARRAY_AGG(validatorindex), '{}') AS validators FROM sync_committees WHERE period IN(?) AND validatorindex IN(?)`, periods, validators)
+		query, args, err := sqlx.In(`
+			SELECT COALESCE(ARRAY_AGG(validatorindex), '{}') AS validators
+			FROM sync_committees
+			WHERE period IN (?) AND validatorindex IN (?)
+			GROUP BY period
+			ORDER BY period DESC
+		`, periods, validators)
 		if err != nil {
 			return types.SyncCommitteesStats{}, err
 		}
-		err = db.ReaderDb.Get(&validatorsInSyncCommittees, db.ReaderDb.Rebind(query), args...)
+		err = db.ReaderDb.Select(&syncCommitteeValidators, db.ReaderDb.Rebind(query), args...)
 		if err != nil {
 			return types.SyncCommitteesStats{}, err
 		}
 
-		if len(validatorsInSyncCommittees.Validators) > 0 {
-			// get and add up2date sync committee statistics from bigtable
+		// if there validators are present in relevant periods, query bigtable
+		if len(syncCommitteeValidators) > 0 {
+			// flatten validator list
 			vs := []uint64{}
-			for _, v := range validatorsInSyncCommittees.Validators {
-				vs = append(vs, uint64(v))
+			for _, scv := range syncCommitteeValidators {
+				for _, v := range scv.Validators {
+					vs = append(vs, uint64(v))
+				}
 			}
 
-			syncStats, err := db.BigtableClient.GetValidatorSyncCommitteesStats(vs, lastExportedEpoch, epoch)
+			// get sync stats from bigtable
+			res, err := db.BigtableClient.GetValidatorSyncDutiesHistory(vs, lastExportedEpoch, epoch)
 			if err != nil {
 				return retv, fmt.Errorf("error retrieving validator sync participations data from bigtable: %v", err)
 			}
+			// add sync stats for validators in latest period
+			latestPeriodCount := len(syncCommitteeValidators[0].Validators)
+			syncStats := utils.AddSyncStats(vs[:latestPeriodCount], res, nil)
+			// if latest sync period is the active one, add remaining scheduled slots
+			if utils.FirstEpochOfSyncPeriod(periodOfEpoch+1)-1 >= services.LatestEpoch() {
```

Comment:
This check does not work like the one here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/48d6141def39f9f188e3e5a8722677ff7cae707f/handlers/validator.go#L818

Here is what I did (everything is working with my local Prater instance using your branch):
1) Navigate to http://localhost:8081/validator/9917
This validator was part of the last sync committee and said committee has not been fully exported yet.
The following information is shown:
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/f78f7322-98dc-442b-95f8-ba137b727343)
Mind the 0 scheduled slots.
2) I have then started the app locally and connected it with my local Prater instance
3) I have only added validator 9917 and viewed the app's dashboard
3) The app shows scheduled slots:
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/25b27c4b-7e9d-4224-8de0-2047cf9c806a)

While the explorer does NOT trigger
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/48d6141def39f9f188e3e5a8722677ff7cae707f/handlers/validator.go#L819
the app DOES trigger
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/48d6141def39f9f188e3e5a8722677ff7cae707f/handlers/api.go#L1174

Please check. If you need more information, I will gladly try to provide it to you. If you need me to perform tests with the app, hit me up as well.

Reply:
Seems to work fine now, thank you!